packages/validate/src/lib/validation.ts

```diff
@@ -9,9 +9,26 @@ import {
 
 /** @public */
 export type ValidatorFn<T> = (value: unknown) => T
+/** @public */
+export type ValidatorUsingKnownGoodVersionFn<In, Out = In> = (
+	knownGoodValue: In,
+	value: unknown
+) => Out
```

Comment:
is it worth adding tests to the validator library for this change?

Reply:
working on tests now